From 26b67b018bcc7008c956a7102d2d902f59c549b5 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 12 Jan 2018 13:32:35 +0000
Subject: [PATCH 0893/1077] MANTIS 0031697 change: allow to reboot kiosk while
 application is in SAM idle

mControl.exe
V1.0.2.21

- added implementation of setKioskStatus


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1301 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/Controller/inc/mcontrol_def.h          | 11 ++++++++---
 .../Massai/cpp/Controller/src/server.cpp              |  3 ++-
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Controller/inc/mcontrol_def.h b/SelfServiceCommon/Massai/cpp/Controller/inc/mcontrol_def.h
index 7d255aae..5750e9b0 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/inc/mcontrol_def.h
+++ b/SelfServiceCommon/Massai/cpp/Controller/inc/mcontrol_def.h
@@ -9,8 +9,8 @@
  *
  ************************************************************************/
 
-#define FILE_VER_BIN     1,0,2,20
-#define FILE_VER_STR     "FileVersion", "1.0.2.20"
+#define FILE_VER_BIN     1,0,2,21
+#define FILE_VER_STR     "FileVersion", "1.0.2.21"
 
 /*!
 \file
@@ -18,12 +18,17 @@
 
 \page mControl.exe Release Notes - mControl.exe - platform startup, starts mudpate.exe, mstarter.exe; servant for systemControl interface
 
-@version 1.0.2.20
+@version 1.0.2.21
 
 \section change1 Change history
 
 mControl.exe
 
+\date 12.01.2018
+\li V1.0.2.21
+\li change - SteT Mantis 0031697 mcontrol: allow to reboot kiosk while application is in SAM idle
+- added implementation of setKioskStatus
+
 \date 22.03.2017
 \li V1.0.2.20
 \li change - SteT Mantis 0044645: PLATFORM.REBOOT.DelayRange generates a new random value every minute
diff --git a/SelfServiceCommon/Massai/cpp/Controller/src/server.cpp b/SelfServiceCommon/Massai/cpp/Controller/src/server.cpp
index 48a97f13..897082bf 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/src/server.cpp
+++ b/SelfServiceCommon/Massai/cpp/Controller/src/server.cpp
@@ -985,6 +985,7 @@ CORBA::Long controllerServant::executeCommand( const Controller::kioskControl::s
 //---------------------------------------------------------------------------
 void controllerServant::setKioskStatus(Controller::kioskStatus status)
 {
-  TRACE(HERE, "not implemented, because it's not used in this version of mControl");
+    fKioskBusy = (status == Controller::IDLE)?FALSE:TRUE;
+    TRACE(HERE,"Kiosk busy: %s", noYes(fKioskBusy));
 }
 
-- 
2.41.0.windows.1

